music_assistant/providers/dlna/player.py

```diff
@@ -219,13 +219,15 @@ async def set_dynamic_attributes(self) -> None:
         _device_uri = self.device.current_track_uri or ""
         self.set_current_media(uri=_device_uri, clear_all=True)
 
+        # Default to MA queue; only override if we detect a known external source
+        self._attr_active_source = self.player_id
         if "spotify" in _device_uri:
+            # Spotify or Spotify Connect
             self._attr_active_source = "spotify"
-        elif _device_uri.startswith("http"):
+        elif _device_uri and not _device_uri.startswith(self.mass.streams.base_url):
+            # Device is playing from an external HTTP source (not MA stream)
             self._attr_active_source = "http"
```

Comment:
why not flip this logic ?

if _device_uri and _device_uri.startswith(self.mass.streams.base_url):
    self._attr_active_source = None  # MA active or unknown

Reply:
Flipped!